src/core/providers/ogr/qgsgeopackageproviderconnection.cpp

```diff
@@ -263,6 +265,21 @@ void QgsGeoPackageProviderConnection::setDefaultCapabilities()
 #endif
 }
 
+void QgsGeoPackageProviderConnection::enableAmphibiousMode( const QString &uri )
```

Comment:
I doubt this method does anything useful. The SQL execute here will only be valid during the lifetime of the SQLite connectino, that is the gdal::ogr_datasource_unique_ptr hDS object, which is destroyed at the end of this method.

Reply:
Yeah what I noticed was that this made new connections to the gpkg, that connection had amphibious mode activated but this was not the main connection where things occured. I initially made this since I was not too sure how to catch an existing connection and then call the desired function. I was hoping that it would still work somehow but alas I was wrong.

I am not too sure how connections are represented, store and how a query can be passed to them as I could not find such examples in this or the ogr provider connection. I'll have to dig further and possibly change more things to do what I aimed to do.

Reply:
@rouault what confuses me is the fact that I replicated in part executeGDalSqlPrivate. I'll try with the connection release.

Reply:
Ok with release connection it removes most issues, though I am left with 1 extra connection, which is confusing. I also need to test the behaviour but this will be hard to do as python is broken (and so are plugins) in my builds...